File: timetable-backend/app/models/conflict.py
from sqlalchemy import Column, Integer, String
from app.database import Base

class Conflict(Base):
    __tablename__ = "conflicts"

    id = Column(Integer, primary_key=True)
    type = Column(String)        # faculty / room / section
    entity_id = Column(Integer)
    timeslot_id = Column(Integer)
    reason = Column(String)

File: timetable-backend-mern/models/Conflict.js
const mongoose = require('mongoose');

const conflictSchema = new mongoose.Schema({
    type: {
        type: String,
        enum: ['faculty', 'room', 'section'],
        required: true,
    },
    entityId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
    },
    timeslotId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'TimeSlot',
        required: true,
    },
    reason: {
        type: String,
        required: true,
    },
    proposalId: {
        type: Number,
        required: false
    }
}, {
    timestamps: true,
});

module.exports = mongoose.model('Conflict', conflictSchema);
